### PR TITLE
Ldap commands replaced with ldap library calls

### PR DESCRIPTION
**Pull request by udayan-y21**
# Backend

# Problem Statement

- _Overview: JIRA number/GitHub Issue added to PR: [Y/N]:_
- _Describe the problem this patch intends to solve._

## Design

- _For Bug describe the fix here._
- _For feature, post the link to the solution page._
- _Any northbound interface change and has been notified to other gatekeepers [Y/N]:_

## Coding

- _Coding conventions are followed and code is consistent. Yes/No?_
- _Confirm All CODACY errors are resolved. Yes/No?_

## Testing

- _Confirm that Test Cases are added (for both the cases, fix and feature). Yes/No?_
- _Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability. Yes/No?_
- _Confirm Testing was performed with installed RPM. Yes/No?_
- _Confirm all the ut/regression/smoke test has been performed [Y/N]_

## Checklist

- _PR is self-reviewed? Yes/No?_
- _GitHub Issue is updated_
- _Jira is updated_
  -  _Check if the description is clear and explained._ 
    -  _Check Acceptance Criterion is defined._
      -  _All the tests performed should be mentioned before Resolving a JIRA._
        -  _Verification needs to be done before marked as Closed/Verified_
        - _Any interface change Yes/No?_
        - _Change notified to other gatekeepers: Yes/No?_
        - _Code reviews done and addressed: Yes/No?_
        - _Codacy issues reviewed and addressed: Yes/No?_
        - _Deployment test : Done/Not?_
        - _UT/smoke test: Done/Not?_
        - _Jira number added to PR: Yes/No?_
        - _Github merge done using UI: Yes/No?_
        - _Side effects on other features - deployment/update/node-replacement_
        - _Changes to quick-start-guide/community impact_
        - _All dependent component code PR are raised or already merged Yes/No_
        - _All dependent code already merged in landing branch Yes/No_

### CODE REVIEW

File: csm/conf/configure.py [hunk 1]
```diff
@@ -324,16 +325,14 @@ def _configure_csm_ldap_schema(self):
         ldap_url = Setup._get_ldap_url()
```
Comment (rohitkolapkar) [inline csm/conf/configure.py:325]: Please check if 'ldap_url' is required ?

File: csm/conf/configure.py [hunk 2]
```diff
@@ -371,16 +369,13 @@ def _create_csm_ldap_user(self):
         ldap_user = Conf.get(const.CSM_GLOBAL_INDEX, const.LDAP_AUTH_CSM_USER)
         csm_ldap_secret =  Conf.get(const.CSM_GLOBAL_INDEX, const.LDAP_AUTH_CSM_SECRET)
         csm_admin_ldap_password = self._fetch_ldap_password(csm_ldap_secret)
-        _output, _err, _rc = self._run_ldap_cmd(f'slappasswd -s {csm_admin_ldap_password}')
-        csm_admin_ldap_password = _output.split('\n')[0]
         ldap_url = Setup._get_ldap_url()
```
Comment (rohitkolapkar) [inline csm/conf/configure.py:372]: Please check if 'ldap_url' is required ?

File: csm/conf/configure.py [hunk 3]
```diff
@@ -371,16 +369,13 @@ def _create_csm_ldap_user(self):
...
         tmpl_init_data = Text(const.CSM_LDAP_ADMIN_USER_LDIF).load()
         tmpl_init_data = tmpl_init_data.replace('<base-dn>',base_dn)
         tmpl_init_data = tmpl_init_data.replace('<csm-admin-user>',ldap_user)
         tmpl_init_data = tmpl_init_data.replace('<csm-admin-password>',csm_admin_ldap_password)
         Text(const.CSM_LDAP_ADMIN_FILE_PATH).dump(tmpl_init_data)
-        self._run_ldap_cmd(f'ldapadd -x -D {bind_base_dn} -w {_rootdnpassword} -f {const.CSM_LDAP_ADMIN_FILE_PATH}\
-        -H {ldap_url}')
+        self._perform_ldif_parsing(const.CSM_LDAP_ADMIN_FILE_PATH, bind_base_dn, _rootdnpassword)
 
     def _run_ldap_cmd(self, cmd):
```
Comment (rohitkolapkar) [inline csm/conf/configure.py:380]: Please check if this function required now ?

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]

**Review by pawankumarsrivastava — APPROVED** [review]

File: csm/conf/configure.py (hunk 3 above)
Comment (udayan-y21) [reply csm/conf/configure.py]: Modifications done

File: csm/conf/configure.py (hunk 2 above)
Comment (udayan-y21) [reply csm/conf/configure.py]: Modifications done

File: csm/conf/configure.py (hunk 1 above)
Comment (udayan-y21) [reply csm/conf/configure.py]: Modifications done

**Review by rohitkolapkar — APPROVED** [review]

**Pull request merged**